### DISCUSSION TITLE
Handle many requests on the same Stream

### DISCUSSION THREAD

**Discussion by moezRebai** [rabbitmq/rabbitmq-stream-dotnet-client#361] [Q&A]
Hello,

Is it possible for RabbitMq stream to handle many requests on the same stream ?

Our case :

- Spot Price publisher that receive many request (for different currencies) and publish spot price updates for each request
- Consumer that send many requests to the broker to retrieve currencies spot price updates ( request for each currency)

The question is it possible to have one stream **_currency-spot-stream_** that handle all the requests ? Or we need to create a separate stream for each request ?

How RabbitMq will handle the synchronization between the producer and the consumer for each request ?

Regards,
Moez

**Comment by lukebakken** [comment]
@moezRebai thanks for using RabbitMQ.

* What have you tried to accomplish your use-case? If you have code, please share it in runnable form.
* If you have not yet written code, how have you tried to answer this question yourself? If our documentation is not adequate, it would be great for us to know so we can improve it.

**Reply by moezRebai** [reply]
Thanks for the answer 😀. I was asking about the concept, is it handled in the library or not yet or is there something that can help to start with

**Reply by lukebakken** [reply]
OK, it seems like you don't have any code.

What about trying to answer the question for yourself using our documentation?
* https://rabbitmq.github.io/rabbitmq-stream-dotnet-client/stable/htmlsingle/index.html
* https://www.rabbitmq.com/streams.html
* https://blog.rabbitmq.com/posts/2023/10/stream-filtering/ (new upcoming feature)

**Comment by Gsantomaggio** [comment]
Thanks. @lukebakken 

>Is it possible for RabbitMq stream to handle many requests on the same stream ?

Yes.

>Consumer that send many requests to the broker to retrieve currencies spot price update

Stream consumers don't send requests. Consumers receive the messages in push. 


>The question is it possible to have one stream currency-spot-stream that handle all the requests ?

I don't fully understand your use case, but yes you can have multiple producers per stream.

 
I'd add these resources:
 -  https://www.youtube.com/watch?v=aRNUYAeJdOw 
 - Getting started: https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/tree/main/docs/Documentation

**Reply by moezRebai** [reply]
hello,

Thanks for your answer, And I already read the documentation ( thanks for the video, it gives me more explanations)

However in my case, producer streams generation are triggred by consumer requests like in the attached image :

![image](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/assets/12102463/240f9689-c302-421f-b535-51f07ea1461c)

Please let me know if you need more details on my case

**Reply by lukebakken** [reply]
@Gsantomaggio and I are happy to help you _help yourself_.

Your diagram makes little sense to me. If you'd like to continue receiving assistance, I suggest writing some code for us to review, or at the very least, provide MUCH more specific details about what you're trying to accomplish.

**Reply by Gsantomaggio** [reply]
> thanks for the video, it gives me more explanations

ok great :) 

>Please let me know if you need more details on my case

What is the question here ?

**Reply by moezRebai** [reply]
@lukebakken What I m trying to achieve is Remote Procedure Call (RPC) on Rabbit Stream, and I m working on it, and i m not sure if stream-filtring can help me

**Reply by lukebakken** [reply]
RabbitMQ streams is probably not the correct tool for you to use for RPC, especially not a single stream.

**Comment by Gsantomaggio** [comment]
Closed due to inactivity 
@moezRebai let me know if there are news.
